yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -332,7 +332,18 @@ export class P2PClient<T extends P2PClientType = P2PClientType.Full>
    * Uses the batched Request Response protocol to request a set of transactions from the network.
    */
   public async requestTxsByHash(txHashes: TxHash[]): Promise<(Tx | undefined)[]> {
-    const txs = await this.p2pService.sendBatchRequest(ReqRespSubProtocol.TX, txHashes);
+    // Set collective timeout based on the number of txs we are requesting
+    const timeoutMs = Math.min(8000, txHashes.length * 100);
```

Comment:
Slight concern over this. From the looks of how this all works, if I need say 2 txs, we would only allow 200ms for the full round trip. Is that correct?

I think that is too low. I think there is a fixed time delay to any request, regardless of how large, for the request to go across the world, the recipient to read from their db and the response to come all the way back again. Given that these requests should in general be for small numbers of transactions I'm worried this change could make things worse.

I think at the very least it should be something like

```
const timeoutMs = Math.min(8000, Math.max(txHashes.length * 100, 1000));
```